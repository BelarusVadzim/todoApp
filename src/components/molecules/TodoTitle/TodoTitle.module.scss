@import "src/variables.scss";

.todoTitle {
  color: white;
}

