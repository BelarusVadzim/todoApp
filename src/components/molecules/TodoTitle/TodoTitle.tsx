import React from 'react';
import { Title } from 'components/atoms/Title';
import { PropsWithChildren } from 'react';
import style from './TodoTitle.module.scss';
import { PropsWithClassName } from 'types';

type TodoTitleProps = PropsWithChildren & PropsWithClassName;

const TodoTitle: React.FC<TodoTitleProps> = ({ children, className }) => {
  const combinedClasses = className
    ? `${style.todoTitle} ${className}`
    : style.todoTitle;

  return (
    <div className={combinedClasses}>
      <Title>{children}</Title>
    </div>
  );
};

export default TodoTitle;
