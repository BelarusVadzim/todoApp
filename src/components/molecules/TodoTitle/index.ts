import TodoTitle from './TodoTitle';

export { TodoTitle };
