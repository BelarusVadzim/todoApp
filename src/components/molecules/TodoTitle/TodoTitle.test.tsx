import { render } from '@testing-library/react';
import TodoTitle from './TodoTitle';

jest.mock('./TodoListFooter.module.scss', () => ({
  todoListFooter: 'todoListFooterStyle',
}));

jest.mock('components/atoms', () => ({
  Title: jest.fn(({ children }) => {
    return <div>{children}</div>;
  }),
}));

describe('<TodoTitle />', () => {
  it('should render properly', () => {
    const component = render(<TodoTitle />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when className specified should render properly', () => {
    const component = render(<TodoTitle className="testClassName" />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
