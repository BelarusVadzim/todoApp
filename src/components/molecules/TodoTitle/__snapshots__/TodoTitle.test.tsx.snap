// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTitle /> should render properly 1`] = `
<DocumentFragment>
  <div>
    <span />
  </div>
</DocumentFragment>
`;

exports[`<TodoTitle /> when className specified should render properly 1`] = `
<DocumentFragment>
  <div
    class="undefined testClassName"
  >
    <span />
  </div>
</DocumentFragment>
`;
